### PR TITLE
Jeremy branch

### PR DESCRIPTION
**Pull request by JAGMAN248**
Added JoystickMap class (Allows easier xbox button mapping), IntakeAndHold class (Pacman mode), and I cleaned up my code.

### CODE REVIEW

File: src/org/usfirst/frc/team2706/robot/JoystickMap.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public static final int XBOX_POV_DOWN_RIGHT = 4;
+    public static final int XBOX_POV_DOWN = 5;
+    public static final int XBOX_POV_DOWN_LEFT = 6;
+    public static final int XBOX_POV_LEFT = 7;
+    public static final int XBOX_POV_UP_LEFT = 8;
+    
+
+    public JoystickMap() {
+        
+        // TODO Auto-generated constructor stub
```
Comment (Vanilla-Rain) [inline src/org/usfirst/frc/team2706/robot/JoystickMap.java]: Remove this, if not the whole constructor

File: src/org/usfirst/frc/team2706/robot/JoystickMap.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package org.usfirst.frc.team2706.robot;
+
+public class JoystickMap {
```
Comment (Vanilla-Rain) [inline src/org/usfirst/frc/team2706/robot/JoystickMap.java]: Looks like a cool idea! Maybe make joystick maps for different controllers, eg Xbox Controller, Logitech Controller, Normal Joystick, then do like JoystickMap.XBox.BACK_LEFT_TRIGGER or something

File: src/org/usfirst/frc/team2706/robot/OI.java [hunk 3]
```diff
@@ -25,6 +28,7 @@
 
     // Joystick for controlling the mechanisms of the robot
     private final Joystick controlStick;
+    
 
     public Joystick getDriverJoystick() {
```
Comment (Vanilla-Rain) [inline src/org/usfirst/frc/team2706/robot/OI.java]: We need to completely rework the OI mappings on Friday, and I also did a little bit of  mapping in a different branch which well add in then

File: src/org/usfirst/frc/team2706/robot/commands/EjectCube.java [hunk 4]
```diff
@@ -41,8 +44,6 @@ public EjectCube(Supplier<Double> speed) {
         inhale = Robot.intake;
         
         this.speed = speed;
-        
-        this.requires(Robot.intake);
```
Comment (Vanilla-Rain) [inline src/org/usfirst/frc/team2706/robot/commands/EjectCube.java]: Does it not require intake?

File: src/org/usfirst/frc/team2706/robot/commands/IntakeAndHold.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * I don't believe initialization is required 
+     */
+    public void initialize() {}
+    
+    /**
+     * Turns the motors on to suck in the cube
+     */
+    public void execute() {
+        System.out.println(inhale.readIRSensor());
```
Comment (Vanilla-Rain) [inline src/org/usfirst/frc/team2706/robot/commands/IntakeAndHold.java]: Use Log.d() instead, its our logging system

**Comment by JAGMAN248** [comment]
Ok, Thanks for checking!

File: src/org/usfirst/frc/team2706/robot/commands/IntakeAndHold.java (hunk 5 above)
Comment (JAGMAN248) [reply src/org/usfirst/frc/team2706/robot/commands/IntakeAndHold.java]: Ok, I fixed it. I didn't know how the logging system worked...

File: src/org/usfirst/frc/team2706/robot/JoystickMap.java (hunk 2 above)
Comment (JAGMAN248) [reply src/org/usfirst/frc/team2706/robot/JoystickMap.java]: Yes, that is a good plan.

**Pull request merged**